engines/account/app/controllers/account/api/v1/login_controller.rb

```diff
@@ -16,7 +17,7 @@ def login
         private
 
         def login_params
-          params.permit(:email, :password)
+          params.permit(:username, :email, :password)
```

Comment:
```ruby
  context = ::Account::User
  user = if params[:username]                     
           context.where(username: params[:username])
         else
           context.where(email: params[:email])      
         end.first
```

You can even move the `if` to a nicer method with early returns. The main point is that OR SQL statements doesn't take advantage of indexes and/or generally just make hard to pick them.

Another thing is that `users` table still doesn't have indexes :joy_cat: